packages/gasket-core/lib/index.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-console, no-process-env */
+// @ts-check
+/// <reference types="./index" />
+
+import GasketEngine from '@gasket/engine';
+import { applyConfigOverrides } from '@gasket/utils';
+
+/**
+ * Get the environment to use for the gasket instance.
+ * Defaults to `development`.
+ * @returns {string} env
+ */
+function getEnvironment(
+  // flags, commandId, warn
+) {
+  // if (flags.env) {
+  //   debug('Environment was passed through command line flags', flags.env);
+  //   return flags.env;
+  // }
+
+  const { GASKET_ENV } = process.env;
+  if (GASKET_ENV) {
+    return GASKET_ENV;
+  }
+
+  // // special snowflake case to match up `local` env with command unless set
+  // if (commandId === 'local') {
+  //   debug('Environment defaulting to `local` due to `local` command');
+  //   return 'local';
+  // }
+
+  const { NODE_ENV } = process.env;
+  if (NODE_ENV) {
+    console.warn(`No env specified, falling back to NODE_ENV: "${NODE_ENV}".`);
+    return NODE_ENV;
+  }
+
+  console.warn('No env specified, falling back to "development".');
+  return 'development';
+}
+/* eslint-enable no-console, no-process-env */
+
+
+function registerActions(instance) {
+  const actions = {};
+  const actionPluginMap = {};
+
+  instance.execApplySync('actions', async (plugin, handler) => {
+    const results = handler(); // The gasket parameter is automatically applied
+    if (results) {
+      Object.keys(results).forEach(actionName => {
+        if (actionPluginMap[actionName]) {
+          // eslint-disable-next-line no-console
+          console.error(
+            `Action '${actionName}' from '${plugin.name}' was registered by '${actionPluginMap[actionName]}'`
```

Comment:
Is there any concern of naming collisions in the future between plugins that have actions with the same name?

It looks like this line takes care of the problem with actions having the same name but I'm wondering in the future when more actions are created if we will run into this issue more often and then we have to make a name change and a publish to fix it. 

We could potentially look into aliasing action names? `gasket.actions.<plugin-short-name>.<action-name>`. It's uglier but could help with collisions. 

This might not be a problem but just wanted to bring up this thought to see what the team thinks.

Reply:
Good question. I think it'll be important that the action names be unique. Aliases _could_ work possibly in application code, but we will not be able to indicate them properly with Types for intellisense and type checks. Also, some plugins will actually execute the actions of other plugins they depend on, in which case they will not know about any aliasing.

Reply:
If anything, we may consider adding a 'strict' option or something that hard fails if duplicate action names are registered. That or make it the default behavior.